pkg/operator/operator.go

```diff
@@ -158,7 +158,9 @@ func (optr *Operator) handleErr(err error, key interface{}) {
 		optr.queue.AddRateLimited(key)
 		return
 	}
-
+	if err := optr.statusFailing(err.Error()); err != nil {
```

Comment:
Are there any cases where `err != nil` and it's still fine not to set the status failing?

Reply:
There could be any transient errors like because of network connectivity in such cases should be retried before setting operator status.